portfolio/src/main/webapp/about.js

```diff
@@ -0,0 +1,93 @@
+let answer = 0;
+
+function loadInTruth() {
+
+    let firstTruth = Math.floor(Math.random() * truths.length);
+    let secondTruth = Math.floor(Math.random() * truths.length);
+    while (firstTruth == secondTruth && truths.length != 1) {
+        secondTruth = Math.floor(Math.random() * truths.length);
+    }
+
+    let firstPlace = Math.floor(Math.random() * 3);
+    let secondPlace = Math.floor(Math.random() * 3);
+    while (firstPlace == secondPlace) {
+        secondPlace = Math.floor(Math.random() * 3);
+    }
+
+    let truthLocations = new Array(2);
+    truthLocations[0] = firstPlace;
+    truthLocations[1] = secondPlace;
+
+    if (firstPlace==0) {
+        document.getElementById("first_txt").innerHTML = truths[firstTruth][0];
+        document.getElementById("first_img").src = truths[firstTruth][1];
+    } else if (firstPlace==1) {
+        document.getElementById("second_txt").innerHTML = truths[firstTruth][0];
+        document.getElementById("second_img").src = truths[firstTruth][1];
+    } else if (firstPlace==2) {
+        document.getElementById("third_txt").innerHTML = truths[firstTruth][0];
+        document.getElementById("third_img").src = truths[firstTruth][1];
+    }
+
+    if (secondPlace == 0) {
+        document.getElementById("first_txt").innerHTML = truths[secondTruth][0];
+        document.getElementById("first_img").src = truths[secondTruth][1];
+    } else if (secondPlace==1) {
+        document.getElementById("second_txt").innerHTML = truths[secondTruth][0];
+        document.getElementById("second_img").src = truths[secondTruth][1];
+    } else if (secondPlace==2) {
+        document.getElementById("third_txt").innerHTML = truths[secondTruth][0];
+        document.getElementById("third_img").src = truths[secondTruth][1];
+    }
+
+    return truthLocations;
+}
+
+function loadInLie(truthLocations) {
+    //determine the location without a truth value
+    let lieLocation = (3 - (truthLocations[0] + truthLocations[1]));
+
+    let lie = Math.floor(Math.random() * lies.length);
+    
+    if (lieLocation==0) {
+        document.getElementById('first_txt').innerHTML = lies[lie][0];
+        document.getElementById('first_img').src = lies[lie][1];
+    } else if (lieLocation==1) {
+        document.getElementById('second_txt').innerHTML = lies[lie][0];
+        document.getElementById('second_img').src = lies[lie][1];
+    } else if (lieLocation==2) {
+        document.getElementById('third_txt').innerHTML = lies[lie][0];
+        document.getElementById('third_img').src = lies[lie][1];
+    }
+    return lieLocation;
+}
+
+function loadInNewQuestion() {
+    let truthLocations = loadInTruth();
+    answer = loadInLie(truthLocations);
+}
+
+function next()  {
```

Comment:
To avoid confusion, without having seen your comment Kenzie, I said to Andrew in conversation that I wasn't sure about the order.  My reasoning was just that I can never remember the cases in which javascript allows code to call other code that hasn't been defined yet.  Since the next() method calls the other methods in the file, I was thinking that if the next() method was moved to the top of the file, the JS might complain when it tries to call other methods that aren't defined until further down.

But if the JS does allow everything to work in that order, I do agree that in most code, we tend to put the important / first-used methods at the top and the private / helper methods below.

Reply:
Ok, just resolve this if you figure it out, Andrew :)

Reply:
Changed to top down